src/network/connection.cpp

```diff
@@ -2089,12 +1981,57 @@ void * ConnectionReceiveThread::Thread()
 	return NULL;
 }
 
+bool ConnectionReceiveThread::wait(int timeout_us, fd_set &readset) {
```

Comment:
Yes.  If it's being modified it should be passed as a pointer.
If it's more advantageous syntax-wise to treat it as a reference, there's nothing stopping you from doing T &fooref = *fooarg;